src/compositions/useVariables.ts

```diff
@@ -0,0 +1,54 @@
+import { useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+import { Ref, computed, ref } from 'vue'
+import { useCan } from '@/compositions/useCan'
+import { useWorkspaceApi } from '@/compositions/useWorkspaceApi'
+import { VariablesFilter } from '@/models'
+import { WorkspaceVariablesApi } from '@/services'
+import { MaybeRef } from '@/types'
+import { UseEntitySubscription } from '@/types/useEntitySubscription'
+import { isNullish } from '@/utilities'
+
+export type UseVariables = UseEntitySubscription<WorkspaceVariablesApi['getVariables'], 'variables'>
+
+export function useVariables(filter: MaybeRef<VariablesFilter>): UseVariables
+export function useVariables(variableIds: MaybeRef<string[] | null | undefined>): UseVariables
+export function useVariables(filter?: MaybeRef<string[] | VariablesFilter | null | undefined>): UseVariables {
+  const filterRef: Ref<string[] | VariablesFilter | null | undefined> = ref(filter)
+
+  const api = useWorkspaceApi()
+  const can = useCan()
+
+  const variablesFilter = computed<[VariablesFilter] | null>(() => {
+    if (!can.read.variable) {
+      return null
+    }
+
+    if (isNullish(filterRef.value)) {
+      return [{}]
+    }
+
+    if (Array.isArray(filterRef.value)) {
```

Comment:
Nice. I like this. Going to have to use this for the `useFlowRuns` compositions.